flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -434,15 +445,12 @@ public int getBuffersInBacklog() {
 		}
 	}
 
-	private boolean shouldNotifyDataAvailable() {
-		// Notify only when we added first finished buffer.
-		return readView != null && !flushRequested && getNumberOfFinishedBuffers() == 1;
+	private boolean mayNotifyDataAvailable() {
+		return readView != null && !dataAvailableNotified;
 	}
 
 	private void notifyDataAvailable() {
```

Comment:
we can remove this method completely, because calling this method has no difference with calling `readView.notifyDataAvailable()` directly.

Reply:
Refer to https://github.com/apache/flink/pull/11786/files#r410367542, we can keep this method to add the `dataAvailableNotified = true` inside.